src/components/OfflineSearch.ts

```diff
@@ -0,0 +1,174 @@
+import { Component, createElement } from "react";
+import { findDOMNode } from "react-dom";
+
+import * as dijitRegistry from "dijit/registry";
+
+import { Alert } from "./Alert";
+import "../ui/OfflineSearch.css";
+
+type SearchMethodOptions = "equals" | "contains";
+
+interface ListView extends mxui.widget._WidgetBase {
+    _datasource: {
+        _constraints: string | null;
+        _setSize: number;
+        atEnd: () => boolean;
+        _pageSize: number;
+    };
+    _loadMore: () => void;
+    _onLoad: () => void;
+    _renderData: () => void;
+    update: () => void;
+}
+
+interface OfflineSearchProps {
+    defaultQuery: string;
+    searchEntity: string;
+    searchAttribute: string;
+    showSearchBar: boolean;
+    targetGridName: string;
+    searchMethod: SearchMethodOptions;
+}
+
+interface OfflineSearchState {
+    alertMessage?: string;
+    buttonVisibility?: string;
+}
+
+export default class OfflineSearch extends Component<OfflineSearchProps, OfflineSearchState> {
+    // internal variables
+    private searchButton: HTMLButtonElement;
+    private searchInput: HTMLInputElement;
+    private targetWidget: ListView;
+
+    constructor(props: OfflineSearchProps) {
+        super(props);
+
+        this.state = {
+            alertMessage: "",
+            buttonVisibility: "hidden"
+        };
+        this.onClear = this.onClear.bind(this);
+        this.onSearchKeyDown = this.onSearchKeyDown.bind(this);
+    }
+
+    render() {
+        return (
+        createElement("div", { className: "widget-offline-search" },
```

Comment:
Incorrect indentation. Either move to previous line, or indent to the right. Prefer to have this on the same line as `return`, without the brackets.